### ISSUE TITLE
Rotate in discrete steps on mobile phone

### ISSUE THREAD

**Issue by mapbakery** [excalidraw/excalidraw#1351]
I love the rotate in discrete steps feature (by constraining the rotation by holding the SHIFT key) but I would like to use that on my phone too. Is it possible to add some kind for setting for that so that I can define the degrees which must be used when rotating? Or I would be happy with a setting for 22.5, 30 or 45 degrees steps.

A solution without a special UI setting would be great although I don't know what would be possible. Perhaps rotating closer tot the center (as mentioned as a response to my tweet). Or always use small discrete steps when using a touch interface, e.g. small steps of 7.5 degrees would also be acceptable.

**Comment by dai-shi** [comment]
copy&pasting my comment in twitter:
> (It would be nice to have discrete steps on mobile too. I wish there would be a way without introducing any UI controls. Any ideas? If it's very close to the center, it rotates in steps? Yeah, worth trying.)

**Comment by dai-shi** [comment]
> Or always use small discrete steps when using a touch interface, e.g. small steps of 7.5 degrees would also be acceptable.

You might want to share your use case. What kind of diagram you would like to draw on mobile. An example would be great.

> small steps of 7.5 degrees

This sounds like you would prefer having grids too? #521

**Comment by mapbakery** [comment]
In my diagrams I use transit map techniques to keep diagrams more readable by restricting the diagonals to certain angles, like in the attached examples. Snap to grid doesn't solve that because that would only work horizontally and vertically unless the grid also contains diagonals.
![transitmapexample](https://user-images.githubusercontent.com/63105006/78971632-52f78a80-7b0c-11ea-9f5c-960a7688dce3.jpg)
![SFTPpattern](https://user-images.githubusercontent.com/63105006/78971635-53902100-7b0c-11ea-836a-7ab2d9cabc96.png)

**Comment by mapbakery** [comment]
Now I think if it, diagonal lines and arrows can be restricted to 22.5 degrees steps by pressing SHIFT while drawing the or arrow. So that is indeed different from rotating. But this is probably not so easy to implement, unless by using a UI setting.

**Comment by dwelle** [comment]
Yes, arrows/lines can rotated in that fashion by dragging the corner handle with `shift`. I don't think we want to get rid of that behavior even when we implement rotation handles for arrows, because rotating arrows by corner handles is more intuitive and easy.

/cc @dai-shi

**Comment by mapbakery** [comment]
So we need some kind of SHIFT key behaviour for mobile use. Which reminds me that I have a bluetooth keyboard. And when i use that (I'm testing it now) I can use the SHIFT key although it is unpractical. And I most of the times I don't have the keyboard with me.

**Comment by dwelle** [comment]
> So we need some kind of SHIFT key behaviour for mobile use. 

Yes. I meant keeping `Shift` on desktop.

**Comment by 2ni** [comment]
I see multiple cases on mobile where shift/cmd/ctrl/alt would be needed. Eg how do you create a circle (not an ellipse) on the mobile version without the shift key, how do you enlarge a existing rectangle by keeping the centre (with ALT on desktop), how to draw straight lines (with shift on Desktop), ...

How Affinity designer solves it:
- for some commands you can just put your left finger on the screen and with the right hand holding the pencil you can draw. This behaves as if you hold the shift button
- they also have a little additional ui with alt, cmd button which you can put your left finger on to imitate an additional key press while drawing

As an idea to better draw without a keyboard?
